Sequences/MRML/vtkMRMLSequenceNode.cxx

```diff
@@ -256,7 +259,19 @@ void vtkMRMLSequenceNode::SetDataNodeAtValue(vtkMRMLNode* node, const char* inde
     return;
   }
 
+  std::string newNodeName = this->SequenceScene->GetUniqueNameByString( node->GetName() );
   vtkMRMLNode* newNode=this->SequenceScene->CopyNode(node);
+  newNode->SetName( newNodeName.c_str() ); // Make sure all the node names in the sequence's scene are unique for saving purposes
+  
+  // Hack to enforce deep copying of volumes (otherwise, a separate image data is not stored for each frame)
+  // TODO: Is there a better way to architecture this?
+  vtkMRMLVolumeNode* volNode = vtkMRMLVolumeNode::SafeDownCast(newNode);
+  if (volNode!=NULL)
+  {
+    vtkImageData* imageDataCopy = vtkImageData::New();
```

Comment:
Even better, updating the `Copy` function of the sequence node similarly to what is done in the `vtkMRMLTableNode` would be ideal I think.
See https://github.com/Slicer/Slicer/blob/34ca8e58f784d2721e95792c564182dce1a44842/Libs/MRML/Core/vtkMRMLTableNode.cxx#L111-L133